### PR TITLE
refac: Cleanup dependency injections

### PR DESCRIPTION
**Pull request by dstenroejl**
## Description

- [x] Cleanup Logging DI registrations
- [x] Remove `SystemDateTimeProvider` and substitute with `IClock`

Deployed to `dev_002`: 
- Before merge with main: https://github.com/Energinet-DataHub/dh3-environments/actions/runs/10737893272
- After merge with main (outbox): https://github.com/Energinet-DataHub/dh3-environments/actions/runs/10739446713

## References

https://app.zenhub.com/workspaces/mosaic-60a6105157304f00119be86e/issues/gh/energinet-datahub/team-mosaic/293

## Checklist
- [ ] Should the change be behind a feature flag?
- [ ] Can the feature be meaningfully disabled or circumvented if there are issues (e.g., database-breaking changes)?
- [ ] Has it been considered whether data is being delivered to the wrong actor?
- [x] Subsystem test executed (dev_002/dev_003)
- [ ] Is there time to monitor state of the release to Production?
- [x] Reference to the task

### CODE REVIEW

File: source/IncomingMessages.Application/UseCases/ReceiveIncomingMarketMessage.cs [hunk 1]
```diff
@@ -12,40 +12,39 @@
...
+    private readonly ValidateIncomingMessage _validateIncomingMessage;
+    private readonly ResponseFactory _responseFactory;
+    private readonly IArchivedMessagesClient _archivedMessagesClient;
+    private readonly ILogger<IncomingMessageClient> _logger;
+    private readonly IIncomingMessageReceiver _incomingMessageReceiver;
+    private readonly DelegateIncomingMessage _delegateIncomingMessage;
+    private readonly IClock _clock;
+
+    public ReceiveIncomingMarketMessage(
+     MarketMessageParser marketMessageParser,
```
Comment (dstenroejl) [inline source/IncomingMessages.Application/UseCases/ReceiveIncomingMarketMessage.cs:40]: Fix indention

File: source/IncomingMessages.Application/UseCases/ReceiveIncomingMarketMessage.cs [hunk 2]
```diff
@@ -54,15 +53,15 @@ public ReceiveIncomingMarketMessage(
...
     }
 
-       public async Task<ResponseMessage> ReceiveIncomingMarketMessageAsync(
-        IIncomingMarketMessageStream incomingMarketMessageStream,
-        DocumentFormat incomingDocumentFormat,
-        IncomingDocumentType documentType,
-        DocumentFormat responseDocumentFormat,
-        CancellationToken cancellationToken)
+    public async Task<ResponseMessage> ReceiveIncomingMarketMessageAsync(
+     IIncomingMarketMessageStream incomingMarketMessageStream,
```
Comment (dstenroejl) [inline source/IncomingMessages.Application/UseCases/ReceiveIncomingMarketMessage.cs:60]: Fix indention

File: source/IncomingMessages.Application/UseCases/ReceiveIncomingMarketMessage.cs [hunk 3]
```diff
@@ -126,19 +125,19 @@ await _delegateIncomingMessage
         return _responseFactory.From(result, responseDocumentFormat);
     }
 
-       private async Task ArchiveIncomingMessageAsync(
-        IIncomingMarketMessageStream incomingMarketMessageStream,
-        IIncomingMessage incomingMessage,
-        IncomingDocumentType incomingDocumentType,
-        CancellationToken cancellationToken)
+    private async Task ArchiveIncomingMessageAsync(
+     IIncomingMarketMessageStream incomingMarketMessageStream,
```
Comment (dstenroejl) [inline source/IncomingMessages.Application/UseCases/ReceiveIncomingMarketMessage.cs:129]: Fix indention

File: source/IntegrationTests/Application/IncomingMessages/GivenIncomingMessagesIsReceivedWithDelegationTests.cs [hunk 4]
```diff
@@ -49,7 +49,7 @@ public GivenIncomingMessagesIsReceivedWithDelegationTests(IntegrationTestFixture
         _senderSpy = new ServiceBusSenderSpy("Fake");
         _serviceBusClientSenderFactory.AddSenderSpy(_senderSpy);
         _incomingMessagesRequest = GetService<IIncomingMessageClient>();
-        _dateTimeProvider = (SystemDateTimeProviderStub)GetService<ISystemDateTimeProvider>();
+        _dateTimeProvider = (SystemDateTimeProviderStub)GetService<IClock>();
```
Comment (dstenroejl) [inline source/IntegrationTests/Application/IncomingMessages/GivenIncomingMessagesIsReceivedWithDelegationTests.cs:52]: Should we rename places like this to e.g. `_clock` and `ClockStub?`

**Review by dstenroejl — COMMENTED** [review]
Commented

File: source/IntegrationTests/Application/IncomingMessages/GivenIncomingMessagesIsReceivedWithDelegationTests.cs [hunk 5]
```diff
@@ -33,7 +33,7 @@ namespace Energinet.DataHub.EDI.IntegrationTests.Application.IncomingMessages;
 
 public class GivenIncomingMessagesIsReceivedWithDelegationTests : TestBase
 {
-    private readonly SystemDateTimeProviderStub _dateTimeProvider;
+    private readonly ClockStub _dateTimeProvider;
```
Comment (MWO1024) [inline source/IntegrationTests/Application/IncomingMessages/GivenIncomingMessagesIsReceivedWithDelegationTests.cs:36]: Should we rename the variable name?

File: source/IntegrationTests/Application/IncomingMessages/WhenIncomingMessagesIsReceivedTests.cs [hunk 6]
```diff
@@ -46,7 +47,7 @@ public class WhenIncomingMessagesIsReceivedTests : TestBase
     private readonly ServiceBusSenderFactoryStub _serviceBusClientSenderFactory;
     private readonly ServiceBusSenderSpy _senderSpy;
     private readonly IncomingMessagesContext _incomingMessageContext;
-    private readonly SystemDateTimeProviderStub _dateTimeProvider;
+    private readonly ClockStub _dateTimeProvider;
```
Comment (MWO1024) [inline source/IntegrationTests/Application/IncomingMessages/WhenIncomingMessagesIsReceivedTests.cs:50]: Should we rename the variable name?

File: source/IntegrationTests/Application/OutgoingMessages/WhenAPeekIsRequestedTests.cs [hunk 7]
```diff
@@ -46,15 +46,15 @@ public class WhenAPeekIsRequestedTests : TestBase
     private readonly EnergyResultPerEnergySupplierPerBalanceResponsibleMessageDtoBuilder _energyResultPerEnergySupplierPerBalanceResponsibleMessageDtoBuilder;
     private readonly EnergyResultPerGridAreaMessageDtoBuilder _energyResultPerGridAreaMessageDtoBuilder;
     private readonly IOutgoingMessagesClient _outgoingMessagesClient;
-    private readonly SystemDateTimeProviderStub _dateTimeProvider;
+    private readonly ClockStub _dateTimeProvider;
```
Comment (MWO1024) [inline source/IntegrationTests/Application/OutgoingMessages/WhenAPeekIsRequestedTests.cs:49]: Should we rename the variable name?

File: source/IntegrationTests/Application/OutgoingMessages/WhenEnqueueingOutgoingMessageWithDelegationTests.cs [hunk 8]
```diff
@@ -39,7 +39,7 @@ public class WhenEnqueueingOutgoingMessageWithDelegationTests : TestBase
     private readonly EnergyResultPerGridAreaMessageDtoBuilder _energyResultPerGridAreaMessageDtoBuilder;
     private readonly IOutgoingMessagesClient _outgoingMessagesClient;
     private readonly ActorMessageQueueContext _context;
-    private readonly SystemDateTimeProviderStub _dateTimeProvider;
+    private readonly ClockStub _dateTimeProvider;
```
Comment (MWO1024) [inline source/IntegrationTests/Application/OutgoingMessages/WhenEnqueueingOutgoingMessageWithDelegationTests.cs:42]: Should we rename the variable name?

File: source/IntegrationTests/Behaviours/IncomingRequests/GivenB2CAggregatedMeasureDataRequestTests.cs [hunk 9]
```diff
@@ -193,7 +193,7 @@ private async Task<string> GivenReceivedAggregatedMeasureDataRequest(
     {
         var incomingMessageClient = GetService<IIncomingMessageClient>();
         var dateTimeZone = GetService<DateTimeZone>();
-        var systemDateTimeProvider = GetService<ISystemDateTimeProvider>();
+        var systemDateTimeProvider = GetService<IClock>();
```
Comment (MWO1024) [inline source/IntegrationTests/Behaviours/IncomingRequests/GivenB2CAggregatedMeasureDataRequestTests.cs:196]: Should we rename the variable name?

File: source/IntegrationTests/Infrastructure/Retention/RemoveOldGridAreaOwnersWhenADayHasPassedTests.cs [hunk 10]
```diff
@@ -186,7 +186,7 @@ private async Task<ActorNumber> GetGridAreaOwnersForGridArea(string gridAreaCode
         return await _masterDataClient.GetGridOwnerForGridAreaCodeAsync(gridAreaCode, CancellationToken.None);
     }
 
-    private sealed class SystemProviderMock : ISystemDateTimeProvider
+    private sealed class SystemProviderMock : IClock
```
Comment (MWO1024) [inline source/IntegrationTests/Infrastructure/Retention/RemoveOldGridAreaOwnersWhenADayHasPassedTests.cs:189]: Should we rename the class?

File: source/IntegrationTests/TestBase.cs [hunk 11]
```diff
@@ -277,8 +278,8 @@ private Task ProcessReceivedInboxEventsAsync()
 
     private Task ProcessBackgroundTasksAsync()
     {
-        var datetimeProvider = GetService<ISystemDateTimeProvider>();
-        return GetService<IMediator>().Publish(new TenSecondsHasHasPassed(datetimeProvider.Now()));
+        var datetimeProvider = GetService<IClock>();
```
Comment (MWO1024) [inline source/IntegrationTests/TestBase.cs:281]: Should we rename the variable?

**Review by MWO1024 — APPROVED** [review]

File: source/IntegrationTests/Infrastructure/Retention/RemoveOldGridAreaOwnersWhenADayHasPassedTests.cs (hunk 10 above)
Comment (dstenroejl) [reply source/IntegrationTests/Infrastructure/Retention/RemoveOldGridAreaOwnersWhenADayHasPassedTests.cs:189]: Refactored instead

**Review by MWO1024 — APPROVED** [review]

**Pull request merged**